packages/opencensus-core/src/common/console-logger.ts

```diff
@@ -92,22 +93,31 @@ class ConsoleLogger implements types.Logger {
    */
   // tslint:disable:no-any
   silly(message: any, ...args: any[]): void {
-    this.logger.silly(arguments);
+    this.logger.silly(util.format(message, ...args));
   }
 }
 
-let defaultLogger = null;
 
 /**
  *  Function logger exported to others classes.
  * @param options A logger options or strig to logger in console
  */
-const logger = (options?: types.LoggerOptions|string) => {
-  defaultLogger = new ConsoleLogger(options);
-  logger['logger'] = defaultLogger;
-  return defaultLogger;
-};
+const logger = (options?: types.LoggerOptions|string|number) => {
+  let opt: types.LoggerOptions|string;
+  if (typeof options === 'number') {
+    if (options < 0) {
+      options = 0;
+    } else if (options > ConsoleLogger.LEVELS.length) {
+      options = ConsoleLogger.LEVELS.length - 1;
+    }
+    opt = {level: ConsoleLogger.LEVELS[options]};
+  } else {
+    opt = options;
+  }
 
-logger();
+  const aLogger = new ConsoleLogger(opt);
+  logger['logger'] = aLogger;
```

Comment:
If you call `logger()` multiple times, you get different loggers. If that is desired, I don't see why you need to store the logger on the `['logger']` property. If that is not desired, you should add something like `if (logger['logger']) {...}` to avoid re-creating the object each time.

Reply:
If we always return the same logger instance, will not be possible to have loggers with different log levels at the same time, what can limit some test capability. I will add a TODO for reevaluation.